packages/components/src/menu-item/types.d.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { IconType } from '../icon';
+
+export type MenuItemProps = {
+	/**
+	 * A CSS `class` to give to the container element.
+	 */
+	className?: string;
+	/**
+	 * The children elements.
+	 */
+	children?: ReactNode;
+	/**
+	 * Text to use as description for button text.
+	 */
+	info?: string;
+	/**
+	 * The icon to render. Supported values are: Dashicons (specified as
+	 * strings), functions, Component instances and `null`.
+	 *
+	 * @default null
+	 */
+	icon?: JSX.Element | null;
+	/**
+	 * Determines where to display the provided `icon`.
+	 */
+	iconPosition?: string;
+	/**
+	 * Whether or not the menu item is currently selected.
+	 */
+	isSelected?: boolean;
+	/**
+	 * If shortcut is a string, it is expecting the display text. If shortcut is an object,
+	 * it will accept the properties of `display` (string) and `ariaLabel` (string).
+	 */
+	shortcut?: string | { display: string; ariaLabel: string };
+	/**
+	 * If you need to have selectable menu items use menuitemradio for single select,
+	 * and menuitemcheckbox for multiselect.
+	 *
+	 * @default 'menuitem'
+	 */
+	role?: string;
+	/**
+	 * Allows for markup other than icons or shortcuts to be added to the menu item.
+	 *
+	 */
+	suffix?: ReactNode;
+	/**
+	 * Human-readable label for item.
+	 */
+	label?: string;
+	/**
+	 * On click handler.
+	 */
+	onClick?( event: React.MouseEvent< HTMLButtonElement, MouseEvent > ): void;
+	/**
+	 * On mouse enter handler.
+	 * @default noop
+	 */
+	onMouseEnter?(
+		event: React.MouseEvent< HTMLButtonElement, MouseEvent >
+	): void;
+	/**
+	 * On mouse leave handler.
+	 * @default noop
+	 */
+	onMouseLeave?(
+		event: React.MouseEvent< HTMLButtonElement, MouseEvent >
+	): void;
```

Comment:
```suggestion
```

These can all be removed once we use `WordPressComponentProps` to mix in the standard `button` props.

Reply:
Sorted